File: linguagens_de_programacao/C/qntd_palavra.c
#include <stdio.h>
#include <string.h>

int main() {
    char frase[200];
    char palavra[50];
    int i, j, cont = 0;
    int tamFrase, tamPalavra;

    printf("Digite a frase:\n");
    gets(frase);

    printf("Digite a palavra:\n");
    gets(palavra);

    tamFrase = strlen(frase);
    tamPalavra = strlen(palavra);

    for (i = 0; i <= tamFrase - tamPalavra; i++) {
        int igual = 1;
        for (j = 0; j < tamPalavra; j++) {
            if (frase[i + j] != palavra[j]) {
                igual = 0;
                break;
            }
        }
        if (igual) {
            cont++;
        }
    }

    printf("A palavra '%s' ocorre %d vezes na frase.\n", palavra, cont);

    return 0;
}

File: linguagens_de_programacao/go/qntd_palavra.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var frase, palavra string

	fmt.Print("digite a frase: ")
	fmt.Scanln(&frase)
	fmt.Print("digite a palavra a procurar: ")
	fmt.Scanln(&palavra)

	frase = strings.ToUpper(frase)
	palavra = strings.ToUpper(palavra)

	contador := 0
	for i := 0; i <= len(frase)-len(palavra); i++ {
		if frase[i:i+len(palavra)] == palavra {
			contador++
		}
	}

	fmt.Printf("a palavra %s ocorre %d vezes na frase.\n", palavra, contador)
}
